tests/unit/record-data-test.js

```diff
@@ -198,6 +198,29 @@ module('unit/record-data', function(hooks) {
     assert.equal(rollbackAttributesSpy.getCalls().length, 0, 'rollbackAttributes was not called');
   });
 
+  test('.rollbackAttributes rolls backs the attributes on a base record data when dealing with a projection', function(assert) {
+    assert.expect(1);
+    const projectedRecordData = this.storeWrapper.recordDataFor(
+      'com.bookstore.projected-book',
+      '1'
+    );
+    const baseRecordData = this.storeWrapper.recordDatas[
+      recordDataKey({
+        modelName: 'com.bookstore.book',
+        id: '1',
+      })
+    ];
+    const rollbackAttributesSpy = this.sinon.spy(baseRecordData, 'rollbackAttributes');
+
+    projectedRecordData.rollbackAttributes();
+
+    assert.equal(
```

Comment:
Can you add a test for `isAttrDirty` here please?

Reply:
@igorT I've added a test for `isAttrDirty` here. I thought about removing the one in `record-data` but wanted to get your thoughts on that before doing so. 

Let me know if you'd like to see any other changes.